mantidproject/mantid
# Improve WorkspaceSingleValue interaction in Workbench

**Description of work.**
Interacting with WorkspaceSingleValue in workbench would previously cause a crash. Therefore this PR improves the handling of the `WorkspaceSingleValue` type in workbench, by adding the required axes to the workspace and altering how the `workspacewidget` and workspace of this type interact.

This PR initially set out to rework the `CreateSingleValuedWorkspace` algorithm so that it behaved like a normal workspace (i.e. returned a Workspace2D). But the BinaryOpertationsMD are currently a blocker to this as they expects a WorkspaceSingleValue object. Instead, I just removed the uses of the CreateSingleValuedWorkspace in the C++ code, which should facilitate its removal later down the line. This is motivated by the fact that in most cases the creation of a single valued workspace can be avoided through the use of algorithms such as `Scale`. 




**To test:**
1. Code review
2. To test the changes you can try the following script for example:
```
matrixWS = CreateSampleWorkspace()
singleValuedWorkspace2D = CreateSingleValuedWorkspace(5,2)
```
The following functionality should be present:
1. Double clicking on the `singleWorkspace` should open a table of the data
2. Double clicking on the `matrixWS` should open the normal plotting window (normal behaviour)

And interacting with the WorkspaceSingleValue in any way should no longer crash workbench.


Fixes #28540. 


#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
